### PR TITLE
refactor: use more descriptive bucket names

### PR DESCRIPTION
**Pull request by adlersantos**
## Description

As suggested by @mmht, it's better if we use more descriptive names for GCS buckets such as `public-covid-tracking-project`.

Currently, the bucket names get auto-prefixed with GCP project numbers (e.g. `432165432198-covid19_tracking-dev-destination`). The reason we did this was to ensure they're globally unique across GCS.

This PR implements the following rules:
* If `env != "prod"` the bucket name used gets appended with the env name: `public-covid-tracking-project-[dev|staging]`
* If `env == "prod"` the bucket name used is simply the name itself: `public-covid-tracking-project`

When a bucket name already exists, we simply rename [the bucket name in the config](https://github.com/GoogleCloudPlatform/public-datasets-pipelines/blob/5f8b2afb55b054618061e2a7487d04e95cd7fc8a/samples/dataset.yaml#L78).

## Checklist
- [x] Tests pass.
- [x] Linters pass.
- [x] If this PR adds/edits/deletes a feature, I have updated the [`README`](https://github.com/GoogleCloudPlatform/public-datasets-pipelines/blob/main/README.md) and samples accordingly.

### CODE REVIEW

File: datasets/covid19_tracking/_terraform/covid19_tracking_dataset.tf [hunk 1]
```diff
@@ -24,22 +24,12 @@ output "bigquery_dataset-covid19_tracking-dataset_id" {
   value = google_bigquery_dataset.covid19_tracking.dataset_id
 }
 
-resource "google_storage_bucket" "covid19_tracking-processing" {
-  name                        = "${var.project_num}-covid19_tracking-${var.env}-processing"
+resource "google_storage_bucket" "public-covid-tracking-project" {
+  name                        = "public-covid-tracking-project${var.env == "prod" ? "" : "-${var.env}"}"
   force_destroy               = true
   uniform_bucket_level_access = true
 }
```
Comment (adlersantos) [inline datasets/covid19_tracking/_terraform/covid19_tracking_dataset.tf:31]: This bucket was never used to I'm deleting it.

**Review by leahecole — CHANGES_REQUESTED** [review]
You've already figured this out, but w/o project # prefix you'll get collisions

**Comment by leahecole** [comment]
Also, I don't think you're in danger of hitting this limit yet, but I did double check, the length for a gcs bucket name is 63 characters, so you can still probably be pretty descriptive, have the project number, and be okay, I think 🤞🏻

**Comment by adlersantos** [comment]
I'll close this PR for now while we regroup on a better unique naming strategy (or otherwise, go with @mmht's suggestion of putting the project number at the back).

**Pull request closed without merging**